interface-impl-amf/src/main/java/org/mule/amf/impl/model/ParameterImpl.java

```diff
@@ -258,21 +256,23 @@ private static Boolean needsQuotes(Shape anyShape) {
     return !(NUMBER_DATA_TYPES.contains(dataType) || BOOLEAN_DATA_TYPE.equals(dataType));
   }
 
-  private static boolean hasAnArrayVariant(UnionShape unionShape) {
-    boolean hasAnArrayVariant = false;
+  private static boolean hasShape(Shape shape, Class<?> shapeClass) {
+    return shapeClass.isInstance(shape) ||
+        shape instanceof UnionShape && containsShape((UnionShape) shape, shapeClass);
+  }
+
+  private static boolean containsShape(UnionShape unionShape, Class<?> shapeClass) {
     for (Shape shape : unionShape.anyOf()) {
-      if (shape instanceof ArrayShape) {
-        hasAnArrayVariant = true;
-      } else if (!(shape instanceof NilShape)) {
-        return false;
+      if (shapeClass.isInstance(shape)) {
+        return true;
+      }
+      if (shape instanceof UnionShape) {
+        if (containsShape((UnionShape) shape, shapeClass)) {
+          return true;
+        }
```

Comment:
Maybe?
```suggestion
      if (shape instanceof UnionShape && containsShape((UnionShape) shape, shapeClass)) {
        return true;
```

I don't know if that's better or worse.